### PR TITLE
Replace fms_platform.h with platform_mod in diag_manager

### LINKED ISSUE #375: Removing fms_platform.h from diag_manager modules
**Is your feature request related to a problem? Please describe.**
When there is a bug in the diag manager, the debugger has issues with providing the correct lines/line numbers because of the `#include <fms_platforms.h>` line at the top of the files. 

**Describe the solution you'd like**
Replacing `#include <fms_platforms.h>` with `use platform_mod` and replacing macros INT_KIND, FLOAT_KIND, and DOUBLE_KIND with `i4_kind`, `r4_kind`, and `r8_kind` which are set in platform_mod.

**Describe alternatives you've considered**
As a temporary workaround, I have copied the information from fms_platform.h into the module files.  This is messy and difficult to maintain.

**Additional context**
This type of fix should be added in more places.

### PR DESCRIPTION
**Pull request by thomas-robinson**
**Description**
When there is a bug in the diag manager, the debugger has issues with providing the correct lines/line numbers because of the #include <fms_platforms.h> line at the top of the files

Fixes #375 

**How Has This Been Tested?**
Passes make check (intel and gcc).  I ran AM4 on gaea in debug mode with the DDT debugger to make sure it gets the correct line numbers.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by thomas-robinson** [comment]
This has the changes from #373

File: block_control/block_control.F90 [hunk 1]
```diff
@@ -31,31 +31,31 @@ module block_control_mod
...
   integer :: npz                 !< vertical extent
-  integer, dimension(:),        _ALLOCATABLE :: ibs   _NULL, &  !< block extents for mpp-style
-                                                ibe   _NULL, &  !! decompositions
-                                                jbs   _NULL, &
-                                                jbe   _NULL
-  type(ix_type), dimension(:),  _ALLOCATABLE :: ix    _NULL !< dereference packed index from global index
+  integer, dimension(:),        allocatable :: ibs  , &  !< block extents for mpp-style
+                                                ibe  , &  !! decompositions
+                                                jbs  , &
+                                                jbe
```
Comment (bensonr) [inline block_control/block_control.F90:50]: Minor nit - you munged the original spacing.

File: data_override/data_override.F90 [hunk 2]
```diff
@@ -100,9 +100,9 @@ module data_override_mod
    integer                          :: dims(4)                 ! dimensions(x,y,z,t) of the field in filename
    integer                          :: comp_domain(4)          ! istart,iend,jstart,jend for compute domain
    integer                          :: numthreads
-   real, _ALLOCATABLE               :: lon_in(:) _NULL
-   real, _ALLOCATABLE               :: lat_in(:) _NULL
-   logical, _ALLOCATABLE            :: need_compute(:) _NULL
+   real, allocatable               :: lon_in(:)
+   real, allocatable               :: lat_in(:)
+   logical, allocatable            :: need_compute(:)
```
Comment (bensonr) [inline data_override/data_override.F90:105]: munged original spacing.

**Comment by thomas-robinson** [comment]
> In addition to the minor spacing, is the #include <fms_platform.h> still needed in the non diag_manager files where you've replaced the macros with fortran instrinsics?

I haven't looked into the other modules.  These updates are needed for debugging the second issue in #362.  That's why I started with the diag_manager.  I can continue with the other modules if you think it's worthwhile, but I'm not sure if the otherr modules are using anything else from fms_platforms.h

**Pull request merged**